### ISSUE TITLE
[vLLM] logit_bias silently ignored in vllm_tt Sampler

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3364]
## Issue

`logit_bias` in `vllm.SamplingParams` has no effect on TT hardware. Logit bias values are stored in metadata but never applied to logits before sampling.

**Note**: `test_logit_bias` in `test_sampling_params.py` currently has no `xfail` marker and passes only because it asserts `len(output) > 0` — it does not verify that the bias has any effect. It should be marked `xfail` until this is fixed.

## Root Cause

`XLASupportedSamplingMetadata.logit_bias` is typed as `list[dict[int, float] | None]` — a Python data structure, never converted to a device tensor and never read by the sampler:

```python
# metadata.py
logit_bias: list[dict[int, float] | None] = field(default_factory=lambda: list())
```

`vllm_tt/sampler.py` has no `apply_logit_bias` method. The GPU path uses `LogitBiasLogitsProcessor` (`vllm/v1/sample/logits_processor/builtin.py`) which builds a dense `[batch, vocab]` float32 tensor and applies it as `logits += bias_tensor`.

## Repro

```python
# With bias suppressing token IDs 0–9, output should differ from baseline.
# Currently identical — bias is ignored.
bias = {i: -100.0 for i in range(10)}
params_bias = vllm.SamplingParams(temperature=0.0, max_tokens=16, logit_bias=bias)
params_base = vllm.SamplingParams(temperature=0.0, max_tokens=16)
out_bias = llm.generate(prompt, params_bias)[0].outputs[0].text
out_base = llm.generate(prompt, params_base)[0].outputs[0].text
assert out_bias != out_base  # FAILS — outputs are identical
```

## Suggested Fix

This is the most tractable unimplemented sampler feature — same pattern as `apply_penalties` (#3331):

- In `from_input_batch`, convert `logit_bias` (list of per-request `{token_id: float}` dicts) to a dense `[batch, vocab]` float32 tensor (zeros everywhere, bias values at specified token IDs).
- Add `no_logit_bias: bool = True` gate so the path is absent from compiled graphs when unused.
- Add `apply_logit_bias` to `vllm_tt/sampler.py`: `logits += logit_bias_tensor`.
- Add `@pytest.mark.xfail` to `test_logit_bias` and strengthen its assertion (currently only checks output is non-empty).

**Comment by kmabeeTT** [comment]
Asserts were added via https://github.com/tenstorrent/tt-xla/pull/3367 so this no longer is silently ignored, but fails loudly.
